cranelift-wasm/src/code_translator.rs

```diff
@@ -1028,6 +1028,33 @@ pub fn translate_operator<FE: FuncEnvironment + ?Sized>(
             let (a, b) = state.pop2();
             state.push1(builder.ins().imul(a, b))
         }
+        Operator::I16x8Shl | Operator::I32x4Shl | Operator::I64x2Shl => {
+            let (a, b) = state.pop2();
+            let bitcast_a = optionally_bitcast_vector(a, type_of(op), builder);
+            let bitwidth = i64::from(builder.func.dfg.value_type(a).bits() - 1);
```

Comment:
Please move the `- 1` to the `band_imm`, as the bitwidth is for example 16, not 15.